builtin/providers/aws/resource_aws_dms_replication_task.go

```diff
@@ -235,12 +255,26 @@ func resourceAwsDmsReplicationTaskDelete(d *schema.ResourceData, meta interface{
 
 	_, err := conn.DeleteReplicationTask(request)
 	if err != nil {
+		if dmserr, ok := err.(awserr.Error); ok && dmserr.Code() == "ResourceNotFoundFault" {
+			d.SetId("")
```

Comment:
We should log with `[DEBUG]` that the resource `d.Id()` wasn't found and is being removed from state, so users can at least see in the debug log what/why this is happening